clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -229,6 +229,8 @@ public synchronized ConsumerRecords<K, V> poll(final Duration timeout) {
 
         // update the consumed offset
         final Map<TopicPartition, List<ConsumerRecord<K, V>>> results = new HashMap<>();
+        final Map<TopicPartition, OffsetAndMetadata> nextOffsetAndMetadata = new HashMap<>();
+
```

Comment:
Super nit: remove the extra newline:

```suggestion
        final Map<TopicPartition, OffsetAndMetadata> nextOffsetAndMetadata = new HashMap<>();
```